src/framework/ui/internal/uiengine.cpp

```diff
@@ -143,6 +144,8 @@ void UiEngine::setRootItem(QQuickItem* rootItem)
         return;
     }
 
+    m_engine->setIncubationController(rootItem->window()->incubationController());
```

Comment:
I really doubt that this is necessary. We use `QQmlApplicationEngine`, and in [its docs](https://doc.qt.io/qt-6.2/qqmlapplicationengine.html#details) I see:
> Automatically sets an incubation controller if the scene contains a [QQuickWindow](https://doc.qt.io/qt-6.2/qquickwindow.html).

So I think it's not necessary, and might even interfere with what the engine does automatically. 

Also, a risk of setting it manually is that the QQuickWindow's incubationController is deleted as soon as the window is closed, and I think the window is closed before the engine is destroyed. So at that point the engine will reference an already-deleted incubation controller.

Reply:
You’re right; this is indeed unnecessary.